### PR TITLE
Fix some PDF handing bugs

### PR DESCRIPTION
**Pull request by diracdeltas**
Fix resourceType check for session.webRequest

Test Plan:

~0. Enable payments. Close Brave.~
~1. npm run add-simulated-payment-history~
~2. Go to Brave payments, open your contribution history.~
~3. Click on a PDF file in your contribution history. It should offer to save the file.~

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [x] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

Test Plan:

Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by petemill** [comment]
This PR is against 0.20.x base branch, but would still need to go to master (and 0.21), so I'll cherry-pick to master and push

File: app/pdf.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
 'use strict'
 
 const tabs = require('./browser/tabs')
-const {fileUrl} = require('../../js/lib/appUrlUtil')
+const {fileUrl} = require('../js/lib/appUrlUtil')
```
Comment (diracdeltas) [inline app/pdf.js]: @petemill note that this part doesn't apply to master (the bug doesn't exist in master at all). haven't checked 0.21.x.

**Comment by petemill** [comment]
This PR is based off 0.20.x branch, but the fixes in the _PDFjs.js_ need to be applied to master (and 0.21.x).

The changes in the other file (_pdf.js_) are no longer relevant on master, but they are still needed for 0.21.x.

So this PR may need to be split in to two - one based off master, and one that can apply to 0.21.x and 0.20.x branches.

However, it seems that the reason _PDFjs.js_ is divergent is likely not intentional. I believe it was introduced in https://github.com/syuan100/browser-laptop/commit/05a53eb58c13c645c9db5f1be05e01598ef2822e which should probably have applied to the 0.19.x hotfix, but since the fix was a muon PR, this b-l commit got lost and was never merged down the branch trees to 0.19.x. Will investigate and confirm.

**Comment by petemill** [comment]
Merged #11531 to 0.21 and 0.20 branches which means the changes to _pdf.js_ were no longer required. This PR now only contains one of the changes.

**Comment by petemill** [comment]
@diracdeltas having trouble getting a STR that runs this function - do you have an example? Tried downloading a PDF from google drive but it doesn't go through this logic

**Comment by diracdeltas** [comment]
@petemill i couldn't find STR for this either unfortunately

**Comment by diracdeltas** [comment]
@petemill i'd be fine with marking this no-qa-needed since it's approximately equivalent to a typo fix

**Comment by petemill** [comment]
0.21.x b67a8834b2988d332af0f3b88b3aec865b453c04
0.20.x 6df4865cf799b377669bafda2fee41ccc9eae12a

**Pull request merged**